From 5fd5932f3079cf0d71323fc59c0c2a2dd60fa69e Mon Sep 17 00:00:00 2001
From: Ian Allison <iana@pims.math.ca>
Date: Fri, 9 Jun 2017 20:13:08 +0000
Subject: [PATCH 1/2] adding more headers and shibboleth

Signed-off-by: Ian Allison <iana@pims.math.ca>
---
 nginx.spec | 46 ++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 46 insertions(+)

diff --git a/nginx.spec b/nginx.spec
index 22f0e5e..60546b1 100644
--- a/nginx.spec
+++ b/nginx.spec
@@ -38,6 +38,8 @@ Source103:         404.html
 Source104:         50x.html
 Source200:         README.dynamic
 Source210:         UPGRADE-NOTES-1.6-to-1.10
+Source300:         headers-more-nginx-module
+Source301:         nginx-http-shibboleth
 
 # removes -Werror in upstream build scripts.  -Werror conflicts with
 # -D_FORTIFY_SOURCE=2 causing warnings to turn into errors.
@@ -83,6 +85,7 @@ BuildArch:         noarch
 Requires:          nginx-mod-http-geoip = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-http-image-filter = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-http-perl = %{epoch}:%{version}-%{release}
+Requires:          nginx-mod-http-shibboleth = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-http-xslt-filter = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-mail = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-stream = %{epoch}:%{version}-%{release}
@@ -143,6 +146,22 @@ Requires:          perl(:MODULE_COMPAT_%(eval "`%{__perl} -V:version`"; echo $ve
 %description mod-http-perl
 %{summary}.
 
+%package mod-http-headers-more-filter
+Group:             System Environment/Daemons
+Summary:           Nginx HTTP more headers module
+Requires:          nginx
+
+%description mod-http-headers-more-filter
+%{summary}.
+
+%package mod-http-shibboleth
+Group:             System Environment/Daemons
+Summary:           Nginx HTTP Shibboleth module
+Requires:          nginx
+
+%description mod-http-shibboleth
+%{summary}.
+
 %package mod-http-xslt-filter
 Group:             System Environment/Daemons
 Summary:           Nginx XSLT module
@@ -174,6 +193,8 @@ Requires:          nginx
 %patch0 -p0
 cp %{SOURCE200} .
 cp %{SOURCE210} .
+cp -R -p %{SOURCE300} .
+cp -R -p %{SOURCE301} .
 
 %if 0%{?rhel} < 8
 sed -i -e 's#KillMode=.*#KillMode=process#g' %{SOURCE10}
@@ -226,6 +247,8 @@ export DESTDIR=%{buildroot}
     --with-http_slice_module \
     --with-http_stub_status_module \
     --with-http_perl_module=dynamic \
+    --add-module=%{_sourcedir}/headers-more-nginx-module \
+    --add-dynamic-module=%{_sourcedir}/nginx-http-shibboleth \
     --with-mail=dynamic \
     --with-mail_ssl_module \
     --with-pcre \
@@ -296,6 +319,10 @@ echo 'load_module "%{_libdir}/nginx/modules/ngx_http_image_filter_module.so";' \
     > %{buildroot}%{_datadir}/nginx/modules/mod-http-image-filter.conf
 echo 'load_module "%{_libdir}/nginx/modules/ngx_http_perl_module.so";' \
     > %{buildroot}%{_datadir}/nginx/modules/mod-http-perl.conf
+echo 'load_module "%{_libdir}/nginx/modules/ngx_http_headers_more_filter_module.so";' \
+    > %{buildroot}%{_datadir}/nginx/modules/mod-http-headers-more-filter.conf
+echo 'load_module "%{_libdir}/nginx/modules/ngx_http_shibboleth_module.so";' \
+    > %{buildroot}%{_datadir}/nginx/modules/mod-http-shibboleth.conf
 echo 'load_module "%{_libdir}/nginx/modules/ngx_http_xslt_filter_module.so";' \
     > %{buildroot}%{_datadir}/nginx/modules/mod-http-xslt-filter.conf
 echo 'load_module "%{_libdir}/nginx/modules/ngx_mail_module.so";' \
@@ -328,6 +355,16 @@ if [ $1 -eq 1 ]; then
     /usr/bin/systemctl reload nginx.service >/dev/null 2>&1 || :
 fi
 
+%post mod-http-headers-more-filter
+if [ $1 -eq 1 ]; then
+    /usr/bin/systemctl reload nginx.service >/dev/null 2>&1 || :
+fi
+
+%post mod-http-shibboleth
+if [ $1 -eq 1 ]; then
+    /usr/bin/systemctl reload nginx.service >/dev/null 2>&1 || :
+fi
+
 %post mod-http-xslt-filter
 if [ $1 -eq 1 ]; then
     /usr/bin/systemctl reload nginx.service >/dev/null 2>&1 || :
@@ -343,6 +380,7 @@ if [ $1 -eq 1 ]; then
     /usr/bin/systemctl reload nginx.service >/dev/null 2>&1 || :
 fi
 
+
 %preun
 %systemd_preun nginx.service
 
@@ -415,6 +453,14 @@ fi
 %{perl_vendorarch}/nginx.pm
 %{perl_vendorarch}/auto/nginx/nginx.so
 
+%files mod-http-headers-more-filter
+%{_datadir}/nginx/modules/mod-http-headers-more-filter.conf
+%{_libdir}/nginx/modules/ngx_http_headers_more_filter_module.so
+
+%files mod-http-shibboleth
+%{_datadir}/nginx/modules/mod-http-shibboleth.conf
+%{_libdir}/nginx/modules/ngx_http_shibboleth_module.so
+
 %files mod-http-xslt-filter
 %{_datadir}/nginx/modules/mod-http-xslt-filter.conf
 %{_libdir}/nginx/modules/ngx_http_xslt_filter_module.so
-- 
1.8.3.1


From 22dfb788da2c5f1eaa23f6900309426a2de40a19 Mon Sep 17 00:00:00 2001
From: Ian Allison <iana@pims.math.ca>
Date: Fri, 9 Jun 2017 20:56:34 +0000
Subject: [PATCH 2/2] make headers a dynamic module

Signed-off-by: Ian Allison <iana@pims.math.ca>
---
 nginx.spec | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/nginx.spec b/nginx.spec
index 60546b1..6413798 100644
--- a/nginx.spec
+++ b/nginx.spec
@@ -85,6 +85,7 @@ BuildArch:         noarch
 Requires:          nginx-mod-http-geoip = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-http-image-filter = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-http-perl = %{epoch}:%{version}-%{release}
+Requires:          nginx-mod-http-headers-more-filter = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-http-shibboleth = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-http-xslt-filter = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-mail = %{epoch}:%{version}-%{release}
@@ -247,7 +248,7 @@ export DESTDIR=%{buildroot}
     --with-http_slice_module \
     --with-http_stub_status_module \
     --with-http_perl_module=dynamic \
-    --add-module=%{_sourcedir}/headers-more-nginx-module \
+    --add-dynamic-module=%{_sourcedir}/headers-more-nginx-module \
     --add-dynamic-module=%{_sourcedir}/nginx-http-shibboleth \
     --with-mail=dynamic \
     --with-mail_ssl_module \
-- 
1.8.3.1

